### DISCUSSION TITLE
add sqlmodel support

### DISCUSSION THREAD

**Discussion by grski** [Lolomgrofl/fastapi-genesis#25] [Ideas]
I think it plays out fairly well with fastapi, given it's, well, the same author :)
Example tutorial: https://testdriven.io/blog/fastapi-sqlmodel/ (might be a bit outdated)
https://sqlmodel.tiangolo.com/

I'm happy to help if you are up for it.

The idea here would be to include sqlmodel support in the project (maybe conditional?) to enable fully async interface + better integration with @tiangolo 's / @fastaspi stack.

**Comment by grski** [comment]
@adamjanas
@hkorzeniewski
@LukaszS0110 
@maksmatch 
@oleg2001-1
@piotek8

Nice first Open Source issue, first come first served ;)

**Comment by Lolomgrofl** [comment]
Okay so I was thinking about `SQLModel` before I even started with this project, so I definitely like the idea. 👍🏼 

The reason why I picked `SQLAlchemy` (for now) is that `Pydatnic v2 ` is out and I feel like `SQLAlchemy` is still the standard ORM library atm for Python and FastAPI, and I want to see the power of `SQLAlchemy 2` + `Pydantic v2`. 

But in the longer run, I would definitely like to see `SQLModel` becoming standard in the industry. 🚀

**Comment by Lolomgrofl** [comment]
Also, I'm thinking about creating such a pattern in a future that is totally agnostic of ORM, so basically people can choose which one they want to use. 🤙🏼  Which sounds like a cool and non-trivial upgrade to the existing template and pattern. 💯

**Reply by grski** [reply]
I'll be happy to collaborate on that, already have some ideas in mind how we can achieve that!
In the meantime let's see what we can do about pydantic v2 in sqlmodel for now: https://github.com/tiangolo/sqlmodel/discussions/621
I already started working on it, let's see if I can add it

**Reply by grski** [reply]
https://github.com/tiangolo/sqlmodel/pull/622
